### DISCUSSION TITLE
Show code written within your own organisation

### DISCUSSION THREAD

**Discussion by Sigmanificient** [lowlighter/metrics#539] [Ideas]
Hi, i use a GitHub organization to organize all my school projects.
I am the owner of the organization and the only one member it in.

Unfortunately, the language used stat don't seems to count the repositories within that organization.

> my `language used` stats
![image](https://user-images.githubusercontent.com/53050011/133944462-0024a0c4-08b2-4273-85f1-3dbfaef16a12.png)

> one of the repo in the organization that use `Ada` which ins't listed in my used language.
![image](https://user-images.githubusercontent.com/53050011/133944485-46e79e76-0a3c-468f-8b8a-22635b4813fb.png)

<hr>

The weirdest part about it is that the metrics says i have 58 public repos
![image](https://user-images.githubusercontent.com/53050011/133944526-f4164121-500d-4403-89ec-0aa3a8b96436.png)

But i have only 41 public repos in my profile, so they are definitively count within my public repos
![image](https://user-images.githubusercontent.com/53050011/133944543-4d2f2466-43d4-4e12-b357-f4fd408cf53d.png)

**Comment by lowlighter** [comment]
Hello 👋 

It's probably because `repositories.affiliations` is set to `owner` (default value) while using Metrics insights (which is not configurable).
This option will changes which repositories will be sent back by GitHub API depending on your relationship with it.

When using Metrics embed, it is possible to see your Ada stats if you pass the `collaborator` constraint in addition to `owner`:
![](https://user-images.githubusercontent.com/22963968/134039698-83025629-10af-4e9b-a157-9b165087e13b.png)
*(https://metrics.lecoq.io/Sigmanificient?languages=1&repositories.affiliations=owner,collaborator&languages.limit=0&base=0&languages.details=percentage)*

The reason for this default value is that while it is indeed more accurate for users who use organanizations as "folders", broader affiliations may mess up completely stats for users who uses organizations shared with a lot of people

Also, these stats use the base algorithm, which only check the top repository languages, whether you actually wrote the code or not, so it's not actually that representative of your actual work.

As a side note, Metrics embed (the GitHub action) also offers an "indepth" languages analyzers which will clone your repositories on the GitHub Action runner and use `git log` to actually count the number of bytes you authored, though it's computationnaly more expansive and slower.

Hope it's helpful 🙂

**Reply by Sigmanificient** [reply]
Wow thanks that is good to know!
